src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,282 @@
+                    res = DetermineType(nbline == 0, typeof(bool), res);
+                    ++nbline;
+                    continue;
+                }
+                else
+                {
+                    if (string.IsNullOrEmpty(val))
+                    {
+                        res = DetermineType(nbline == 0, typeof(bool), res);
+                        continue;
+                    }
+                }
+                bool floatParse = float.TryParse(val, out float floatResult);
+                if (floatParse)
+                {
+                    res = DetermineType(nbline == 0, typeof(float), res);
+                    ++nbline;
+                    continue;
+                }
+                else
+                {
+                    if (string.IsNullOrEmpty(val))
+                    {
+                        res = DetermineType(nbline == 0, typeof(float), res);
+                        continue;
+                    }
+                }
+                res = DetermineType(nbline == 0, typeof(string), res);
+                ++nbline;
+            }
+            return res;
+        }
+
+        private static Type DetermineType(bool first, Type suggested, Type previous)
+        {
+            if (first)
+                return suggested;
+            else
+                return MaxKind(suggested, previous);
+        }
+
+        private static Type MaxKind(Type a, Type b)
+        {
+            if (a == typeof(string) || b == typeof(string))
+                return typeof(string);
+            if (a == typeof(float) || b == typeof(float))
+                return typeof(float);
+            if (a == typeof(bool) || b == typeof(bool))
+                return typeof(bool);
+            return typeof(string);
+        }
+
+        /// <summary>
+        /// Reads a text file as a DataFrame.
+        /// Follows pandas API.
+        /// </summary>
+        /// <param name="filename">filename</param>
+        /// <param name="sep">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="names">column names (can be empty)</param>
+        /// <param name="dtypes">column types (can be empty)</param>
+        /// <param name="nrows">number of rows to read</param>
+        /// <param name="guess_rows">number of rows used to guess types</param>
+        /// <param name="encoding">text encoding</param>
+        /// <param name="index">add one column with the row index</param>
+        /// <returns>DataFrame</returns>
+        public static DataFrame ReadCsv(string filename,
+                                char sep = ',', bool header = true,
+                                string[] names = null, Type[] dtypes = null,
+                                int nrows = -1, int guess_rows = 10,
+                                Encoding encoding = null, bool index = false)
+        {
+            return ReadStream(() => new StreamReader(filename, encoding ?? Encoding.ASCII),
```

Comment:
Should this be `Encoding.Default` instead?

Reply:
If the user passes in `null`, I would say we should use the default `StreamReader` constructor - 

[https://source.dot.net/#System.Private.CoreLib/shared/System/IO/StreamReader.cs,180](https://source.dot.net/#System.Private.CoreLib/shared/System/IO/StreamReader.cs,180)

That way `detectEncodingFromByteOrderMarks` gets set as `true`.

---
In reply to: [312262614](https://github.com/dotnet/corefxlab/pull/2710#discussion_r312262614) [](ancestors = 312262614)